plansys2_popf_plan_solver/src/plansys2_popf_plan_solver/popf_plan_solver.cpp

```diff
@@ -50,37 +55,46 @@ POPFPlanSolver::getPlan(
     return {};
   }
 
+  auto output_dir = std::filesystem::path(
+    lc_node_->get_parameter(output_dir_parameter_name_).value_to_string()
+  );
+
   if (node_namespace != "") {
-    std::filesystem::path tp = std::filesystem::temp_directory_path();
     for (auto p : std::filesystem::path(node_namespace) ) {
       if (p != std::filesystem::current_path().root_directory()) {
-        tp /= p;
+        output_dir /= p;
       }
     }
-    std::filesystem::create_directories(tp);
+    std::filesystem::create_directories(output_dir);
   }
+  RCLCPP_INFO(
+    lc_node_->get_logger(), "Writing planning results to %s.", output_dir.string().c_str());
 
   plansys2_msgs::msg::Plan ret;
-  std::ofstream domain_out("/tmp/" + node_namespace + "/domain.pddl");
+
+  const auto domain_file_path = output_dir / std::filesystem::path("domain.pddl");
+  std::ofstream domain_out(domain_file_path);
   domain_out << domain;
   domain_out.close();
 
-  std::ofstream problem_out("/tmp/" + node_namespace + "/problem.pddl");
+  const auto problem_file_path = output_dir / std::filesystem::path("problem.pddl");
+  std::ofstream problem_out(problem_file_path);
   problem_out << problem;
   problem_out.close();
 
-  int status = system(
-    ("ros2 run popf popf " +
-    lc_node_->get_parameter(parameter_name_).value_to_string() +
-    " /tmp/" + node_namespace + "/domain.pddl /tmp/" + node_namespace +
-    "/problem.pddl > /tmp/" + node_namespace + "/plan").c_str());
+  const auto plan_file_path = output_dir / std::filesystem::path("plan");
+  const auto args = lc_node_->get_parameter(arguments_parameter_name_).value_to_string();
```

Comment:
I have seen that references to `node_namespace` in your PR have been removed. Are your changes ready to use namespaces? We must support having different PlanSys2 instances, one in each namespace for not collide

Reply:
Not at all! The namespaces continue to be added in this loop: https://github.com/PlanSys2/ros2_planning_system/blob/2d8ff58b6188a145ad86e2c4e8d510e29534dbd1/plansys2_popf_plan_solver/src/plansys2_popf_plan_solver/popf_plan_solver.cpp#L147-L153

In my tests, the results are definitely written to `<output_dir>/<node_namespace>` (or `<temp_dir>/<node_namespace>` by default)